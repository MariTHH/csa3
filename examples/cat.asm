.data:
    num tmp:

.text:
      in tmp
loop: in tmp
    out_char tmp
	jmp loop
end: hlt
