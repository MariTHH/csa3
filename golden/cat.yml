source: |-
  .data:
      num tmp: 0
  .text:
        in tmp
  loop: in tmp
      out_char tmp
  	  jmp loop
  end: hlt
input: |-
  9, 'G', 'o', 'o', 'd', ' ', 'd', 'a', 'y', '!'
code: |-
  [
      {
          "opcode": "in",
          "addr": 0,
          "args": [
              0
          ]
      },
      {
          "opcode": "in",
          "addr": 1,
          "args": [
              0
          ]
      },
      {
          "opcode": "out_char",
          "addr": 2,
          "args": [
              0
          ]
      },
      {
          "opcode": "jmp",
          "addr": 3,
          "args": [
              1
          ]
      },
      {
          "opcode": "hlt",
          "addr": 4,
          "args": []
      }
  ]
output: |
  source LoC: 16 code instr: 5
  ============================================================
  Good day!
  instr_counter: 28 ticks: 56
log: |
  DEBUG    root:machine.py:277 {TICK: 0, ADDR: 0, PC: 0, OUT: 0, ACC: 0} 
  DEBUG    root:machine.py:138 {TICK: 1, ADDR: 0, PC: 0, OUT: 57, ACC: 57} 
  DEBUG    root:machine.py:138 {TICK: 2, ADDR: 0, PC: 1, OUT: 57, ACC: 57} 
  DEBUG    root:machine.py:138 {TICK: 3, ADDR: 0, PC: 1, OUT: 71, ACC: 71} 
  DEBUG    root:machine.py:138 {TICK: 4, ADDR: 0, PC: 2, OUT: 71, ACC: 71} 
  DEBUG    root:machine.py:138 {TICK: 5, ADDR: 0, PC: 2, OUT: 71, ACC: 71} 
  DEBUG    root:machine.py:100 output: '' << 'G' 
  DEBUG    root:machine.py:138 {TICK: 6, ADDR: 0, PC: 2, OUT: 71, ACC: 71} 
  DEBUG    root:machine.py:138 {TICK: 7, ADDR: 0, PC: 3, OUT: 71, ACC: 71} 
  DEBUG    root:machine.py:138 {TICK: 8, ADDR: 0, PC: 1, OUT: 71, ACC: 71} 
  DEBUG    root:machine.py:138 {TICK: 9, ADDR: 0, PC: 1, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 10, ADDR: 0, PC: 2, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 11, ADDR: 0, PC: 2, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:100 output: 'G' << 'o' 
  DEBUG    root:machine.py:138 {TICK: 12, ADDR: 0, PC: 2, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 13, ADDR: 0, PC: 3, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 14, ADDR: 0, PC: 1, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 15, ADDR: 0, PC: 1, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 16, ADDR: 0, PC: 2, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 17, ADDR: 0, PC: 2, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:100 output: 'Go' << 'o' 
  DEBUG    root:machine.py:138 {TICK: 18, ADDR: 0, PC: 2, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 19, ADDR: 0, PC: 3, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 20, ADDR: 0, PC: 1, OUT: 111, ACC: 111} 
  DEBUG    root:machine.py:138 {TICK: 21, ADDR: 0, PC: 1, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 22, ADDR: 0, PC: 2, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 23, ADDR: 0, PC: 2, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:100 output: 'Goo' << 'd' 
  DEBUG    root:machine.py:138 {TICK: 24, ADDR: 0, PC: 2, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 25, ADDR: 0, PC: 3, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 26, ADDR: 0, PC: 1, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 27, ADDR: 0, PC: 1, OUT: 32, ACC: 32} 
  DEBUG    root:machine.py:138 {TICK: 28, ADDR: 0, PC: 2, OUT: 32, ACC: 32} 
  DEBUG    root:machine.py:138 {TICK: 29, ADDR: 0, PC: 2, OUT: 32, ACC: 32} 
  DEBUG    root:machine.py:100 output: 'Good' << ' ' 
  DEBUG    root:machine.py:138 {TICK: 30, ADDR: 0, PC: 2, OUT: 32, ACC: 32} 
  DEBUG    root:machine.py:138 {TICK: 31, ADDR: 0, PC: 3, OUT: 32, ACC: 32} 
  DEBUG    root:machine.py:138 {TICK: 32, ADDR: 0, PC: 1, OUT: 32, ACC: 32} 
  DEBUG    root:machine.py:138 {TICK: 33, ADDR: 0, PC: 1, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 34, ADDR: 0, PC: 2, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 35, ADDR: 0, PC: 2, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:100 output: 'Good ' << 'd' 
  DEBUG    root:machine.py:138 {TICK: 36, ADDR: 0, PC: 2, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 37, ADDR: 0, PC: 3, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 38, ADDR: 0, PC: 1, OUT: 100, ACC: 100} 
  DEBUG    root:machine.py:138 {TICK: 39, ADDR: 0, PC: 1, OUT: 97, ACC: 97} 
  DEBUG    root:machine.py:138 {TICK: 40, ADDR: 0, PC: 2, OUT: 97, ACC: 97} 
  DEBUG    root:machine.py:138 {TICK: 41, ADDR: 0, PC: 2, OUT: 97, ACC: 97} 
  DEBUG    root:machine.py:100 output: 'Good d' << 'a'
  DEBUG    root:machine.py:138 {TICK: 42, ADDR: 0, PC: 2, OUT: 97, ACC: 97} 
  DEBUG    root:machine.py:138 {TICK: 43, ADDR: 0, PC: 3, OUT: 97, ACC: 97} 
  DEBUG    root:machine.py:138 {TICK: 44, ADDR: 0, PC: 1, OUT: 97, ACC: 97} 
  DEBUG    root:machine.py:138 {TICK: 45, ADDR: 0, PC: 1, OUT: 121, ACC: 121} 
  DEBUG    root:machine.py:138 {TICK: 46, ADDR: 0, PC: 2, OUT: 121, ACC: 121} 
  DEBUG    root:machine.py:138 {TICK: 47, ADDR: 0, PC: 2, OUT: 121, ACC: 121} 
  DEBUG    root:machine.py:100 output: 'Good da' << 'y' 
  DEBUG    root:machine.py:138 {TICK: 48, ADDR: 0, PC: 2, OUT: 121, ACC: 121} 
  DEBUG    root:machine.py:138 {TICK: 49, ADDR: 0, PC: 3, OUT: 121, ACC: 121} 
  DEBUG    root:machine.py:138 {TICK: 50, ADDR: 0, PC: 1, OUT: 121, ACC: 121} 
  DEBUG    root:machine.py:138 {TICK: 51, ADDR: 0, PC: 1, OUT: 33, ACC: 33} 
  DEBUG    root:machine.py:138 {TICK: 52, ADDR: 0, PC: 2, OUT: 33, ACC: 33} 
  DEBUG    root:machine.py:138 {TICK: 53, ADDR: 0, PC: 2, OUT: 33, ACC: 33} 
  DEBUG    root:machine.py:100 output: 'Good day' << '!' 
  DEBUG    root:machine.py:138 {TICK: 54, ADDR: 0, PC: 2, OUT: 33, ACC: 33} 
  DEBUG    root:machine.py:138 {TICK: 55, ADDR: 0, PC: 3, OUT: 33, ACC: 33} 
  DEBUG    root:machine.py:138 {TICK: 56, ADDR: 0, PC: 1, OUT: 33, ACC: 33} 
  WARNING  root:machine.py:287 Input buffer is empty! 
  INFO     root:machine.py:291 output_buffer: 'Good day!' 
